app/components/copy_button/copy_button.css

```diff
@@ -1,11 +1,34 @@
-.CopyButton-success {
-  color: var(--color-green-dark);
+.CopyButton {
+  position: relative;
 }
 
-.CopyButton:not(.CopyButton--success) .CopyButton-success {
+.CopyButton-loading,
+.CopyButton-success {
   display: none;
+
+  position: absolute;
+  top: 50%;
+  left: 0;
+  right: 0;
+  transform: translateY(-50%);
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
 }
 
-.CopyButton--success .CopyButton-label {
-  display: none;
+.CopyButton[data-state='loading'] .CopyButton-label,
+.CopyButton[data-state='success'] .CopyButton-label {
+  visibility: hidden;
+}
+
+.CopyButton[data-state='loading'] .CopyButton-loading {
+  display: initial;
+}
+
+.CopyButton[data-state='success'] .CopyButton-success {
+  display: initial;
+}
+
+.Button:not(.Button--primary) .CopyButton-success {
+  color: var(--color-green-dark);
```

Comment:
Why is this so complicated? Isn't there a way to render HTML conditionally with Stimulus? E.g. [alpine](https://github.com/alpinejs/alpine) can do this easily at runtime, without any integration issues.